src/main/java/com/corundumstudio/socketio/namespace/Namespace.java

```diff
@@ -320,7 +326,7 @@ public void leaveRoom(String room, UUID sessionId) {
         clients.remove(sessionId);
 
         if (clients.isEmpty()) {
-            map.remove(room, Collections.emptySet());
+            map.remove(room, clients);
```

Comment:
What is the reason that replacing Emptyset to clients? @541595308 
It is corrected? @mrniko

Reply:
There are some empty collection after the system has been running for a period of time. I am not sure that it can be deleted by Collections.emptySet(), But this change has no effect

Reply:
It does matter. client.isEmpty and map.remove(room, clients) is not an atomic operation.  There could be a join operation between client.isEmpty and map.remove(room, clients).

Reply:
It could happen that a client just joined a room then you removed the clients set.